src/components/Launch/LaunchWorkflowForm/inputHelpers/datetime.ts

```diff
@@ -1,18 +1,29 @@
 import { dateToTimestamp } from 'common/utils';
 import { Core } from 'flyteidl';
 import { utc as moment } from 'moment';
+import { InputValue } from '../types';
+import { allowedDateFormats } from './constants';
 import { ConverterInput, InputHelper } from './types';
 
+function parseDate(value: InputValue) {
+    return value instanceof Date
+        ? value
+        : moment(value.toString(), allowedDateFormats).toDate();
```

Comment:
I didn't know you could put in an array here -- does it pick one of them preferentially if there's ambiguity to the format?

Reply:
I just learned about it: https://momentjs.com/guides/#/parsing/multiple-formats/
1. If the date isn't in one of the allowed formats, it results in an invalid date, instead of falling back to the JS `Date` constructor
2. It tries them in the order specified.